### PR TITLE
Feature 279/update poll

### PR DESCRIPTION
**Pull request by Maria358**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/components/EditCooperationPollForm.vue [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+        creationDate: { requiredValidator },
+      },
+    };
+  },
+  mounted() {
+    try {
+      this.$store.dispatch(`${StoreModuleEnum.housesStore}/${HousesActionsEnum.SET_HOUSES}`, this.cooperationId);
+      this.initData();
+    } catch {
+      console.log('error was caught during mounting BaseCooperationPoll');
```
Comment (tsehla42) [inline src/components/EditCooperationPollForm.vue:197]: Consider using toast notification here

File: src/components/EditCooperationPollForm.vue [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+      this.pollData.completionDate = this.selectedPoll?.completionDate.toLocaleString('uk-UA');
+
+      this.beginDate = this.selectedPoll?.creationDate;
+      this.finishDate = this.selectedPoll?.completionDate;
+    },
+    async editPoll() {
+      const poll = {
+        header: this.pollData.header,
+        description: this.pollData.description,
+        creationDate: new Date(this.beginDate.toLocaleString('en-US')).toISOString(),
```
Comment (tsehla42) [inline src/components/EditCooperationPollForm.vue:220]: Why do you use en-US locale?

File: src/components/base/BaseCooperationPoll.vue [hunk 3]
```diff
@@ -9,35 +25,159 @@
...
+        return true;
+      }
+      return false;
+    },
+    toggle(event: Event): void {
+      event.stopPropagation();
+      (this.$refs.menu as VueElement & { toggle: (event: Event) => void }).toggle(event);
+    },
+    confirmDeletePoll(event: Event) {
+      if (this.checkStatus(this.$props.poll.status)) return;
```
Comment (tsehla42) [inline src/components/base/BaseCooperationPoll.vue:127]: Please, do not use if clause without { }

File: src/store/polls/mutations.ts [hunk 4]
```diff
@@ -6,6 +6,28 @@ export const mutations: MutationTree<PollsStateInterface> & Mutations = {
...
+  [PollsMutationEnum.DELETE_POLL]: (state, payload) => {
+    const updatePolls = state.cooperationPolls?.filter((el) => el.id !== payload);
+    state.cooperationPolls = updatePolls;
+  },
+
+  [PollsMutationEnum.UPDATE_POLL]: (state, payload) => {
+    state.cooperationPolls?.forEach((el) => {
+      if (el.id === payload.ids.pollId) {
+        el.header = payload.poll.header;
+        el.description = payload.poll.description;
```
Comment (tsehla42) [inline src/store/polls/mutations.ts:23]: I think there is a simpler way to map this

File: src/components/EditCooperationPollForm.vue (hunk 2 above)
Comment (Maria358) [reply src/components/EditCooperationPollForm.vue:220]: Cause constructor of Date accepts only date in En format

**Review by tsehla42 — APPROVED** [review]

**Review by anatoliy-stepanchenko — APPROVED** [review]

**Pull request merged**